pixleesdk/src/main/java/com/pixlee/pixleesdk/ui/widgets/list/BaseRecyclerView.kt

```diff
@@ -98,87 +91,71 @@ open class BaseRecyclerView : RecyclerView {
      *
      * @param pxlKtxAlbum: PXLKtxAlbum? Please pass the same reference that you make photoInfo: PhotoWithVideoInfo with
      * @param pxlWidgetType: String
      */
-    fun enableAutoAnalytics(pxlKtxAlbum: PXLKtxAlbum, pxlWidgetType: String) {
-        this.pxlKtxAlbum = pxlKtxAlbum
-        this.pxlWidgetType = pxlWidgetType
-        fireAnalytics()
-    }
-
-    protected fun fireAnalytics() {
+    var albumForAutoAnalytics: AlbumForAutoAnalytics? = null
+        set(value) {
+            field = value
+            fireOpenAndVisible()
+        }
+    
+    protected fun fireOpenAndVisible() {
         fireAnalyticsOpenedWidget()
-        fireAnalyticsVisibleWidget()
+        fireAnalyticsWidgetVisible()
     }
 
     private var isAnalyticsOpenedWidgetFired: Boolean = false
     private fun fireAnalyticsOpenedWidget() {
-        if (pxlKtxAlbum != null && !isAnalyticsOpenedWidgetFired) {
-            if (pxlWidgetType == null) {
-                Log.e(PXLAnalytics.TAG, "can't fire OpenedWidget analytics event because pxlWidgetType is null")
+        if (PXLClient.autoAnalyticsEnabled && !isAnalyticsOpenedWidgetFired) {
+            if (albumForAutoAnalytics == null) {
+                Log.e(PXLAnalytics.TAG, "can't fire OpenedWidget analytics event because albumForAutoAnalytics is null. Please pass your own AlbumForAutoAnalytics to albumForAutoAnalytics.")
                 return
             }
             if (pxlPhotoAdapter.list.isNotEmpty() && visibility == View.VISIBLE) {
                 isAnalyticsOpenedWidgetFired = true
                 GlobalScope.launch {
-                    pxlKtxAlbum?.also { album ->
-                        pxlWidgetType?.also { pxlWidgetType ->
-                            try {
-                                album.openedWidget(pxlWidgetType)
-                            } catch (e: Exception) {
-                                isAnalyticsOpenedWidgetFired = false
-                                e.printStackTrace()
-                            }
+                    albumForAutoAnalytics?.also {
+                        try {
+                            it.album.openedWidget(it.widgetType)
+                        } catch (e: Exception) {
+                            isAnalyticsOpenedWidgetFired = false
+                            e.printStackTrace()
                         }
                     }
                 }
             }
-
         }
     }
 
-    private var isAnalyticsVisibleWidgetFired: Boolean = false
-    private fun fireAnalyticsVisibleWidget() {
-        if (pxlKtxAlbum != null && !isAnalyticsVisibleWidgetFired) {
-            if (pxlWidgetType == null) {
-                Log.e(PXLAnalytics.TAG, "can't fire WidgetVisible analytics event because pxlWidgetType is null")
+    private var isAnalyticsWidgetVisibleFired: Boolean = false
+    private fun fireAnalyticsWidgetVisible() {
```

Comment:
hm so i see the if statement for autoAnalyticsEnabled for widget visible, and opened widget events, but what about the other events?

Reply:
`openedLightbox`: will be fired in PXLPhotoProductView automatically if PXLClient.autoAnalyticsEnabled is true

There is one exception, `loadMore` has always been fired since the beginning. So, I'm keeping the rule for our existing clients. No mather PXLClient.autoAnalyticsEnabled is true or falue, `loadMore` will be fired when trying to get more photos via PXLAlbum(java) and PXLKtxAlbum(kotlin).

Reply:
if you turn autoAnalyticsEnabled to false, can you fire openedLightbox itself manually?

Reply:
yes, if you set autoAnalyticsEnabled to false, you can fire `openedLightbox` manually as well as `openedWidget`, `widgetVisible`, `actionClicked`, `addToCart`, and `conversion`.